src/bt-gatt-io.c

```diff
@@ -307,17 +307,17 @@ client_register_client_cb(int status, int client_if, bt_uuid_t* app_uuid)
 static void
 client_scan_result_cb(bt_bdaddr_t* bd_addr, int rssi, uint8_t* adv_data)
 {
-  size_t len;
+  size_t len = 0;
   struct pdu_wbuf* wbuf;
 
   if (adv_data) {
-    len = strlen((const char*)adv_data);
+    while (adv_data[len]) {
```

Comment:
|adv_data| is not a string, but packed data like |len1|data1|len2|data2|...|len_n|data_n|? Please add a short comment.